src/mpi/datatype/typeutil.c

```diff
@@ -211,158 +211,93 @@ int MPIR_Datatype_init_predefined(void)
 
 int MPIR_Datatype_builtintype_alignment(MPI_Datatype type)
 {
+    if (type == MPI_DATATYPE_NULL)
+        return 1;
+
```

Comment:
Is this code in a critical path? It's fine if not. Otherwise, just to note that `MPI_DATATYPE_NULL` is an unlikely case.